### DISCUSSION TITLE
How to run in parallel using all CPUs from a laptop?

### DISCUSSION THREAD

**Discussion by iuryt** [CliMA/Oceananigans.jl#3418] [Computational science!]
While discussing with a student at IIT Madras (Chennai, India), she asked me how simple would be to run her simulation in parallel using the cores from her laptop. I just realized that I never tried that and we couldn't find an example on the docs while using the search engine.

I know this should be straightforward, maybe `julia -p nproc simulation.jl`?
Can we specify the number of the cores used within the code?

**Comment by glwagner** [comment]
The easiest way is to use the environment variable `JULIA_NUM_THREADS`. This can be specified in a config file like `.zshrc` or `.bashrc`, or when launching julia:

```bash
$ JULIA_NUM_THREADS=8 julia
```

See https://docs.julialang.org/en/v1/manual/multi-threading/
